File: businesslib/src/main/java/com/wujia/businesslib/event/AbsEventEntiry.java
package com.wujia.businesslib.event;

/**
 * author ：shenbingkai@163.com
 * date ：2019-03-31
 * description ：
 */
public abstract class AbsEventEntiry<T> {

    public AbsEventEntiry() {
    }

    public AbsEventEntiry(IMiessageInvoke<T> mInvoke) {
        this.mInvoke = mInvoke;
    }

    private IMiessageInvoke mInvoke;

    public abstract void onEventBus(T t);

    protected void invoke(T t) {
        if (mInvoke != null)
            mInvoke.eventBus(t);
    }
}

File: businesslib/src/main/java/com/wujia/businesslib/event/AbsEventEntiry.kt
package com.wujia.businesslib.event

/**
 * author ：shenbingkai@163.com
 * date ：2019-03-31
 * description ：
 */
abstract class AbsEventEntiry<T> {

    private lateinit var mInvoke: IMiessageInvoke<T>

    constructor() {}

    constructor(mInvoke: IMiessageInvoke<T>) {
        this.mInvoke = mInvoke
    }

    abstract fun onEventBus(t: T)

    protected operator fun invoke(t: T) {
        mInvoke?.eventBus(t)
    }
}
